src/pages/workspaces/workspace/notebooks/NotebookLauncher.js

```diff
@@ -238,29 +238,31 @@ class NotebookEditor extends Component {
     const { clusters } = this.props
     const prevCluster = getCluster(oldClusters)
     const currCluster = getCluster(clusters)
-    if (prevCluster && prevCluster.id !== currCluster.id) {
+    if (prevCluster && currCluster && prevCluster.id !== currCluster.id) {
       document.location.reload()
     }
   }
 
 
   async startCluster() {
-    const { refreshClusters, ajax: { Jupyter } } = this.props
+    const { refreshClusters, workspace: { workspace: { namespace } }, ajax: { Jupyter } } = this.props
+    await refreshClusters()
+    const cluster = getCluster(this.props.clusters) // Note: reading up-to-date prop
+    if (!cluster) {
+      await Jupyter.cluster(namespace, Utils.generateClusterName()).create({
+        machineConfig: Utils.normalizeMachineConfig({})
+      })
+    }
 
     while (this.mounted) {
       await refreshClusters()
-      const { clusters } = this.props //Note: placed here to read updated value after refresh
-      const cluster = getCluster(clusters)
-      if (!cluster) {
-        return { error: 'You do not have access to run analyses on this workspace.' }
```

Comment:
Is there still some way we can detect this? How does the UI behave if a user can't create a cluster?

Reply:
We shouldn't actually hit this case unless something is misconfigured. If you don't have `canCompute`, you'll get the preview instead.